From b72fea1818e92d18affe88a90f6f2460eab74360 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@freescale.com>
Date: Thu, 5 Nov 2015 15:52:09 +0800
Subject: [PATCH] MLK-11833 bcmdhd: pull back sdpcm_hdrlen length from old skb
 for new skb

Pull back sdpcm_hdrlen length from old skb as new skb here
is passed to postprocessing.
This is a generic fix. We must have it regardless of host is
non scatter-gather or scatter gather capable.

Suggested by: Vinayak Kamath <vinayak.kamath@broadcom.com>
Signed-off-by: Dong Aisheng <aisheng.dong@freescale.com>

(cherry picked from commit 45c96b8881eaa2b073b9dfa2a44229accd7409b7)
---
 drivers/net/wireless/bcmdhd/dhd_sdio.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/dhd_sdio.c b/drivers/net/wireless/bcmdhd/dhd_sdio.c
index 4b628e9..83b65f9 100644
--- a/drivers/net/wireless/bcmdhd/dhd_sdio.c
+++ b/drivers/net/wireless/bcmdhd/dhd_sdio.c
@@ -1877,6 +1877,11 @@ static int dhdsdio_txpkt_preprocess(dhd_bus_t *bus, void *pkt, int chan, int txs
 		PKTALIGN(osh, tmp_pkt, PKTLEN(osh, pkt), DHD_SDALIGN);
 		bcopy(PKTDATA(osh, pkt), PKTDATA(osh, tmp_pkt), PKTLEN(osh, pkt));
 		*new_pkt = tmp_pkt;
+                /*
+		 * pull back sdpcm_hdrlen length from old skb as new skb here
+		 * is passed to postprocessing
+                 */
+		PKTPULL(osh, pkt, sdpcm_hdrlen);
 		pkt = tmp_pkt;
 	}
 
-- 
1.8.0

